src/com/mambu/apisdk/util/RequestExecutorImpl.java

```diff
@@ -966,6 +971,15 @@ private static void logCurlCommandForRequest(Method method, ContentType contentT
 
 	}
 
+	private static String addFullDetailsParam(ParamsMap params, String urlParams) {
+
+		if (params != null && params.size() > 0 && params.get(FULL_DETAILS_QUERY_PARAM) != null) {
```

Comment:
It looks like the second check (_params.size() > 0_) can be removed. If the params is not null, params.get() will not fail. After that, we have two cases. If params contain "fullDetails", we will have only two checks instead of three (params.size() is removed). If params does not contain "fullDetails", we will have also only two checks (as in the current implementation, with .size() in place), only that we replaced .size() with .get().

Reply:
removed it, thanks!